docs/source/frameworks/fastai.rst

```diff
@@ -0,0 +1,246 @@
+=======
+fast.ai
+=======
+
+.. note::
+
+   BentoML requires fastai version 2 or higher to be installed. We do not currently support version 1.
+
+fastai is a deep learning library which provides both high-level components for practioners to get SOTA results in standard deep learning domains, and low-level components
+for researchers to build new approaches. The abstractions can be achieved by leveraging the flexibility of the PyTorch library and its ecosystem (PyTorch Lightning, Catalyst, etc.)
+
+fastai also provides a great sets of `documentation <docs.fast.ai>`_ and
+`migration guides <https://docs.fast.ai/#Migrating-from-other-libraries>`_ from
+PyTorch-related library.
+
+
+Fine tuning the language model
+------------------------------
+
+.. seealso::
+
+   This section is based heavily on `Transfer Learning with text <https://docs.fast.ai/tutorial.text.html#The-ULMFiT-approach>`_ from fastai.
+
+.. code-block:: python
+
+   from fastai.basics import URLs
+   from fastai.metrics import accuracy
+   from fastai.text.data import DataBlock
+   from fastai.text.data import TextBlock
+   from fastai.text.data import untar_data
+   from fastai.text.data import CategoryBlock
+   from fastai.text.models import AWD_LSTM
+   from fastai.text.learner import text_classifier_learner
+   from fastai.data.transforms import parent_label
+   from fastai.data.transforms import get_text_files
+   from fastai.data.transforms import GrandparentSplitter
+
+   # Download IMDB dataset
+   path = untar_data(URLs.IMDB)
+
+   # Create IMDB DataBlock
+   imdb = DataBlock(
+       blocks=(TextBlock.from_folder(path), CategoryBlock),
+       get_items=get_text_files,
+       get_y=parent_label,
+       splitter=GrandparentSplitter(valid_name="test"),
+   )
+   dls = imdb.dataloaders(path)
+
+   # define a Learner object
+   learner = text_classifier_learner(
+        dls, AWD_LSTM, drop_mult=0.5, metrics=accuracy
+    )
+
+   # quickly fine tune the model
+   learner.fine_tune(4, 1e-2)
+
+   # output:
+   # epoch     train_loss  valid_loss  accuracy  time
+   # 0         0.453252    0.395130    0.822080  36:45
+
+We can then test out prediction:
+
+.. code-block:: python
+
+   learner.predict("I really liked that movie!")
+
+   # output:
+   # ('pos', TensorText(1), TensorText([0.1216, 0.8784]))
+
+
+Saving a learner with BentoML
+-----------------------------
+
+   :bdg-warning:`Warning:` `Learner <https://docs.fast.ai/learner.html#Learner>`_ instance is required to save with BentoML.
+   This is a design choice to preserve functionalities provided by fastai.
+
+.. seealso::
+
+   If you want to use the PyTorch model components of fastai ``Learner``s with BentoML, refer to the :ref:`PyTorch Framework Guide<frameworks/pytorch:PyTorch>`.
+
+   To get the PyTorch model, access it via ``learner.model``:
+
+   .. code-block:: python
+
+      import bentoml
+
+      bentoml.pytorch.save_model("my_pytorch_model", learner.model)
```

Comment:
move this section to after the `bentoml.fastai.save_model("fastai_sentiment", learner)` line below?